### ISSUE TITLE
Bus GTFS Modeling for BLoC

### ISSUE THREAD

**Issue by sirmammingtonham** [sirmammingtonham/smartrider#24]
Need to complete integration of the bus GTFS data into the bloc refactor (see [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ) or [here](https://www.youtube.com/watch?v=2zZkVLsySm4) for info about bloc pattern).
Involves creating a repository and provider class to fetch the data, and model classes to decode and turn the data into usable objects. Look at [shuttle bloc](/sirmammingtonham/smartrider/tree/frontend-bloc-refactor/lib/blocs/shuttle) and [shuttle data](/sirmammingtonham/smartrider/tree/frontend-bloc-refactor/lib/data/models/shuttle) for examples.

Static and Realtime (RT) data are modeled differently, see https://gtfs.org/reference/static/ and https://gtfs.org/reference/realtime/v2/

Bus data links:
- GTFS Static Data : https://www.cdta.org/schedules/google_transit.zip
- GTFS RT Data – Trip Updates : http://64.128.172.149:8080/gtfsrealtime/TripUpdates
- GTFS RT Data – Vehicle Positions: http://64.128.172.149:8080/gtfsrealtime/VehiclePositions 
- Service Advisories: https://www.cdta.org/service-advisories/json

Ideally, static data would only be downloaded and unzipped once during the app's first sign-in, and then whenever we push an update to the app. Real-time data should be fetched every couple of seconds, and used to update vehicle positions and route alerts. Probably have to branch off to a separate issue for this, once modeling is finished.

**Comment by jlioanag** [comment]
For the static data, we need to be able to read the txt files that is in the google_transit.zip.

I think this should be good:
https://flutter.dev/docs/cookbook/persistence/reading-writing-files

**Comment by sirmammingtonham** [comment]
added in 
7ff190089d56a4469fcfb5f9b545080a69c00b0f
6542c0498c3817497358f8cdcf899118b6e74cd7
fcc07104f09f45fcdbd9da803d7eadf001354320
d18f268fcbc539d89bdcfe7692fa09c6608a7eb0
882ca48c3b3ca69eafd8d797f5d920d24c8cb4b8
